sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -878,6 +878,14 @@ class Analyzer(
         case _ => None
       }
 
+      val isStreamingTmpView = tmpView.nonEmpty && tmpView.get.isStreaming
```

Comment:
how about
```
tmpView.foreach { v =>
  if (!isStreaming && !SQLConf.get.getConf... && v.isStreaming) ...
  if (isStreaming &&  !v.isStreaming) ...
}
```

Reply:
@cloud-fan @HeartSaVioR Bring back the discussion(https://github.com/apache/spark/pull/30148#discussion_r517785184) here:
After taking a further look, moving the checking logic in the Analyser rule will break both SQL and DataSet API on the reading streaming temp view.
I can only block the streaming temp view reading in the DataSet side, but it breaks the consistency between both APIs, as we don't have streaming keywords in SQL to distinguish the table/view is whether streaming or not. So I just unify the code path and add comments to emphasize the current behavior of `spark.table` and `spark.read.table`.

Reply:
Makes sense to me. It seems overkill if we spend a lot of efforts just to disallow streaming temp view in `spark.read.table`